File: people-among-dragons/src/models/publication.ts
export class Publication {
    _publicationId: number;
    _publicationTitle: string;
    _publicationImage: string;
    _publicationText: string;
    _userName: string;
    _userEmail: string;
    _publicationDate: Date;

    constructor(publicationId:number, publicationTitle: string, publicationImage: string, publicationText: string, userName: string, userEmail: string, publicationDate: Date){
        this._publicationId = publicationId;
        this._publicationTitle = publicationTitle;
        this._publicationImage = publicationImage;
        this._publicationText = publicationText;
        this._userName = userName;
        this._userEmail = userEmail;
        this._publicationDate = publicationDate;
    }
}

File: Dragon-WebApi/Models/Publication.cs
namespace Dragon_WebApi.Models
{
    public class Publication
    {
        public int Id { get; set; }
        public string PublicationTitle { get; set; }
        public string PublicationImage { get; set; }
        public string PublicationText { get; set; }
        public string UserName { get; set; }
        public string UserEmail { get; set; }
        public DateTime PublicationDate { get; set; }

        public Publication()
        {
            PublicationTitle = "";
            PublicationImage = "";
            PublicationText = "";
            UserName = "";
            UserEmail = "";
            PublicationDate = DateTime.Now;
        }
    }
}
